xcode/Subconscious/Shared/Components/Detail/DetailStackView.swift

```diff
@@ -288,61 +287,98 @@ struct DetailStackModel: Hashable, ModelProtocol {
         return Update(state: model)
     }
     
-    static func rebaseLinkOnAddress(
-        address: Slashlink,
-        link: SubSlashlinkLink
-    ) -> Slashlink {
-        // Stitch the base address on to the tapped link, making any
-        // bare slashlinks relative to the sphere they belong to.
-        //
-        // This is needed in the viewer but address will always based
-        // on our sphere in the editor case.
-        guard case let .petname(basePetname) = address.peer else {
-            return link.slashlink
+    public static func findBestAddressForLink(
+        link: SubSlashlinkLink,
+        context: Peer?,
+        environment: TestableEnvironment
+    ) async throws -> Slashlink {
```

Comment:
All the logic for mapping a link to a navigable address is now contained in this function. There are tests for this, but not to full coverage since traversal is impossible in testing.

Did my best to document the thinking, please flag any confusion or tweaks to make.